src/utilities.rs

```diff
@@ -6,26 +6,48 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-#[derive(Debug, Clone, Copy, PartialEq, PartialOrd, Eq, Ord)]
+use rand::{
+    distributions::{Distribution, Standard},
+    Rng,
+};
+use std::fmt::{self, Debug, Formatter};
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, PartialOrd, Eq, Ord)]
 pub struct Name(pub i32);
 
-#[derive(Debug, Clone, Copy, PartialEq, PartialOrd, Eq, Ord)]
-pub struct Age(pub i32);
+impl Distribution<Name> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Name {
+        Name(rng.gen())
+    }
+}
```

Comment:
Running this highlighted that `Name` is signed. I guess it's not a problem but looks a bit funky when printing and you're seeing negative names?